robot-detect

```diff
@@ -186,19 +206,37 @@ enable_fastopen = os.path.exists("/proc/sys/net/ipv4/tcp_fastopen")
 
 try:
     ip = socket.gethostbyname(args.host)
+    server = (ip, args.port)
```

Comment:
Why keep the use of `socket.gethostbyname`?  Wouldn't it make sense to use `socket.getaddrinfo` in all cases, since it works the same way for IPv4 and IPv6?

Additionally, I think that it would enable the script to handle domain names for any type of IP address.

Reply:
Yes, I can also use that. The idea is that `gethostbyname` does not support IPv6, and therefore we only need to do the `getaddrinfo` part if there is no IPv4 host to check in the first place. However, I can simply switch to `getaddrinfo` and use the first (?) result.

Reply:
Yes.  Actually it would probably make more sense to create the socket with `socket.create_connection`, so that you don't have to deal with the list returned by `getaddrinfo`.

Reply:
Yes, I can definitely use that and it eases things as far as creating the socket is concerned, but we still need the IP at least to put it into the CSV not breaking compatibility with previous versions. Whatever I do, I guess I still need to get the IP either using `gethostbyname` (IPv4) or `getaddrinfo` (both, but more convoluted). I would take only the part I need from the current commit and save the IP somewhere, what do you think about it?

Reply:
Could you use `socket.getpeername()` or `socket.getsockname()` to get the information and pass it to other sockets?

Reply:
Done, got it to be much cleaner.